src/content/app/tools/blast/state/general/blastGeneralSlice.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createSlice, type PayloadAction } from '@reduxjs/toolkit';
+
+type BlastView =
+  | 'blast-form' //new job form
+  | 'unviewed-submissions' //unviewed jobs
+  | 'submissions-list' // jobs list (list of viewed)
+  | 'submission-results'; // one submission result
```

Comment:
Regarding the comments:

`new job form` — the form for submitting a new batch of BLAST jobs
`unviewed jobs` — unviewed BLAST submissions
`jobs list (list of viewed)` -> list of viewed BLAST submissions
`one submission result` -> results of BLAST jobs in a single submission

There are two concepts here: a BLAST job is a single 1 sequence vs 1 species BLAST run performed by EBI BLAST. Each of the jobs will receive an id from the EBI BLAST tool.

We can submit one or more jobs at once via the form (which makes the whole thing very confusing). This group of jobs submitted in a single form is referred to in our code as a submission.

What you then see in the lists are the submissions (potentially many sequences against many species at once). And when you press on the "Results" button, you are seeing a page that shows results from multiple BLAST jobs grouped in a single submission.

Reply:
Thanks. I will update the comments....its indeed confusing and i think i will add one more comment of top of this to explain what a job and a submission is.